File: src/main/kotlin/tads/dipas/server/softwares/sapl/Sample.java
package tads.dipas.server.softwares.sapl;

import tads.dipas.server.softwares.img.Image;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "sample")
public class Sample {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    public Long id;
    @Transient
    private int[][][] image;
    @OneToMany
    private List<Seedling> seedlings;
    private String path;
    private double metric;

    public Sample(String path) {
        this.image = Image.imRead(path);
        this.path = path;
        seedlings = new ArrayList<>();
    }

    public Sample() {

    }

    public void setSeedling(Seedling seedling) {
        this.seedlings.add(seedling);
    }

    public Seedling getSeedling(int i) {
        return seedlings.get(i);
    }


    public List<Seedling> getSeedlings() {
        return seedlings;
    }

    public int[][][] getImage() {
        return image;
    }

    public void setImage(int[][][] imagem) {
        this.image = imagem;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String caminho) {
        this.path = caminho;
    }

    public double getMetric() {
        return metric;
    }

    public void setMetric(double metric) {
        this.metric = metric / 10;
    }

    public String getFilename() {
        String[] split = this.path.split("\\.");
        return split[0];
    }

    public double tallestSeedling() {
        double maior = 0;
        for (Seedling p : this.seedlings) {
            if (p.getTotalLength(this.metric) > maior)
                maior = p.getTotalLength(this.metric);
        }
        return maior;
    }
}

File: src/main/kotlin/tads/dipas/server/model/Sample.kt
package tads.dipas.server.model

import org.springframework.hateoas.RepresentationModel
import tads.dipas.server.softwares.sapl.Seedling
import javax.persistence.*

@Entity
class Sample {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    var id: Long? = null

    @OneToMany
    private val images: List<Image>? = null

    @OneToMany
    private val : List<Seedling>? = null
    private val metric = 0.0

}

class SampleResponse(
    var id: Long = -1,
    var date: String = "",
    val creator: String = ""
) : RepresentationModel<ImageResponse>() {

    constructor(image: Image) : this(id = image.id, date = image.date.toString(), creator = image.user.username) {


        // add(linkTo<UserController> {methodOn(UserController::class.java).index()}.withRel("users"))
        // add(linkTo<UserController> {methodOn(UserController::class.java).get(user.id)}.withRel("user"))
    }
}
